WSLBlobNFS/WSLBlobNFS.psm1

```diff
@@ -6,178 +11,539 @@ Set-PSDebug -Strict
+    Execute-WSL "sed -i -e 's/\r$//' $linuxScriptsPath/$queryScriptName"
+
+    Execute-WSL "chmod +x $linuxScriptsPath/$queryScriptName"
+
+    Execute-WSL "ls -l $linuxScriptsPath | grep -E '$wslScriptName|$queryScriptName' > /dev/null 2>&1"
+}
+
+function Intall-WSLBlobNFS-Internal
+{
+    [CmdletBinding()]
+    param()
+
     # Check if wsl is installed or not.
-    $wslstatus = wsl -l -v
-    if($LASTEXITCODE -eq 1)
+    wsl -l -v > $null 2>&1
+    $wslstatus = $LASTEXITCODE
+
+    # Check if the distro is installed or not.
+    # Note: executing a command such as ls is required else if the distro is installed, then control will not come out of wsl command.
+    wsl -d $distroName ls > $null 2>&1
+    $distrostatus = $LASTEXITCODE
+
+    if($wslstatus -eq 1)
     {
         Write-Host "WSL is not installed. Installing WSL..."
 
-        wsl --install -d Ubuntu-22.04
-        Write-Host "Restart the VM to complete WSL installation, and then Run the script again with Initialize-WSLBlobNFS to continue the WSL setup."
+        wsl --install -d $distroName
+        # To-do: Check if the distro is installed successfully or not.
+
+        # To-do: Prompt user to restart the machine to complete WSL installation.
+        Write-Success "Restart the VM to complete WSL installation, and then run Initialize-WSLBlobNFS to continue the WSL setup."
+
+        # Set the exit code to 1 to indicate that the script execution is not completed.
+        $global:LASTEXITCODE = 1
         return
     }
-    Write-Host "WSL is already installed. Skipping WSL installation."
+    elseif($distrostatus -eq -1)
+    {
+        Write-Host "WSL distro $distroName is not installed. Installing WSL distro $distroName..."
+        wsl --install -d $distroName
+        # To-do: Check if the distro is installed successfully or not.
+
+        Write-Success "WSL distro $distroName is installed. Run Initialize-WSLBlobNFS to setup WSL environment for WSLBlobNFS usage."
+
+        # Set the exit code to 1 to indicate that the script execution is not completed.
+        $global:LASTEXITCODE = 1
+        return
+    }
+
     return
 }
 
-function Initialize-WSLBlobNFS
+
+function Initialize-WSLBlobNFS-Internal
 {
-    # Most of the commands require admin privileges. Hence, we need to run the script as admin.
-    $linuxusername = "root"
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $false)]$Force=$false
+    )
 
     # Check if wsl is installed or not.
-    $wslstatus = wsl -l -v
-    if($LASTEXITCODE -eq 1)
+    Intall-WSLBlobNFS-Internal
+    if($LASTEXITCODE -ne 0)
     {
-        Write-Host "WSL is not installed. Installing WSL..."
-
-        wsl --install -d Ubuntu-22.04
-        Write-Host "Restart the VM to complete WSL installation, and then Run the script again with Initialize-WSLBlobNFS to continue the WSL setup."
+        # Set the exit code to 1 to indicate that the script execution is not completed.
+        $global:LASTEXITCODE = 1
```

Comment:
this isn't needed so we can remove this to avoid confusion to the reader and avoid the impression that the developer didn't fully understand variable scoping in PS

Reply:
No Tomar, $global:LASTEXITCODE = 1 is needed to avoid local copy.
https://stackoverflow.com/questions/10943554/powershell-lastexitcode-in-a-function